### PR TITLE
web: add SiteAdminWebhookUpdatePage.

### PR DESCRIPTION
**Pull request by sashaostrikov**
Test plan:
Storybook and manual testing.

Closes https://github.com/sourcegraph/sourcegraph/issues/43490

## App preview:

- [Web](https://sg-web-ao-ui-webhooks-update-webhook.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.27 kb) | 0.04% (+5.42 kb) | 0.04% (+5.15 kb) | 0.40% (+3) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ao/ui/webhooks/update-webhook-part2/compare-report.html;tab=live_object) for a full comparison between the commits 0976bb1 and 61e429d or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/site-admin/SiteAdminWebhookUpdatePage.story.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                                            createExternalService(
+                                                ExternalServiceKind.BITBUCKETSERVER,
+                                                'https://sgdev.bitbucket.org'
+                                            ),
+                                            createExternalService(
+                                                ExternalServiceKind.BITBUCKETSERVER,
+                                                'https://sgprod.bitbucket.org'
+                                            ),
+                                            createExternalService(ExternalServiceKind.GITLAB, 'https://gitlab.com'),
+                                            createExternalService(ExternalServiceKind.PERFORCE, 'https://perforce.com'),
```
Comment (mrnugget) [inline client/web/src/site-admin/SiteAdminWebhookUpdatePage.story.tsx:61]: Can't you use that other help you had for this?

File: client/web/src/site-admin/SiteAdminWebhookUpdatePage.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                            createdBy={data.node.createdBy}
+                            updatedAt={data.node.updatedAt}
+                            updatedBy={data.node.updatedBy}
+                        />
+                    }
+                    className="mb-3"
+                    headingElement="h2"
+                />
+            )}
+            {data?.node && data.node.__typename === 'Webhook' && (
```
Comment (mrnugget) [inline client/web/src/site-admin/SiteAdminWebhookUpdatePage.tsx:54]: Do you really need that typename check?

File: client/web/src/site-admin/WebhookCreateUpdatePage.tsx [hunk 3]
```diff
@@ -238,6 +310,22 @@ function supportedExternalServiceKind(kind: ExternalServiceKind): boolean {
     }
 }
 
+function createUpdateWebhookVariables(webhook: Webhook, id?: string): UpdateWebhookVariables {
```
Comment (mrnugget) [inline client/web/src/site-admin/WebhookCreateUpdatePage.tsx:313]: Nitpick: it's a bit confusing to have "create" and "update" in this name when the page is also called `CreateUpdate`

```suggestion
function buildUpdateWebhookVariables(webhook: Webhook, id?: string): UpdateWebhookVariables {
```

File: client/web/src/site-admin/WebhookCreateUpdatePage.tsx [hunk 4]
```diff
@@ -207,15 +238,56 @@ export const WebhookCreateUpdatePage: FC<WebhookCreateUpdatePageProps> = ({ hist
...
+                                        <Button
+                                            onClick={event => {
+                                                event.preventDefault()
+                                                updateWebhook().catch(
+                                                    // noop here is used because update error is handled directly when useMutation is called
+                                                    noop
+                                                )
+                                            }}
+                                            variant="primary"
+                                            disabled={updateLoading || webhook.name.trim() === ''}
```
Comment (mrnugget) [inline client/web/src/site-admin/WebhookCreateUpdatePage.tsx:253]: Why this? In Ryan's doc it says that `name` is optional. Is it not?

File: client/web/src/site-admin/SiteAdminWebhookUpdatePage.story.tsx (hunk 1 above)
Comment (sashaostrikov) [reply client/web/src/site-admin/SiteAdminWebhookUpdatePage.story.tsx:61]: it is just for the sake of check that it is getting filtered out

File: client/web/src/site-admin/SiteAdminWebhookUpdatePage.tsx (hunk 2 above)
Comment (sashaostrikov) [reply client/web/src/site-admin/SiteAdminWebhookUpdatePage.tsx:54]: not really, we use generics. in some places in our codebase we have them, in some -- we don't. I'll remove it in this case

Comment (mrnugget) [reply client/web/src/site-admin/SiteAdminWebhookUpdatePage.tsx:54]: I think it won't if you pass a type to the function, let me try

File: client/web/src/site-admin/WebhookCreateUpdatePage.tsx (hunk 4 above)
Comment (sashaostrikov) [reply client/web/src/site-admin/WebhookCreateUpdatePage.tsx:253]: @ryanslade I thought that during a discussion with Rob we agreed that we add the name, and it makes sense to make it mandatory. wdyt?

Comment (ryanslade) [reply client/web/src/site-admin/WebhookCreateUpdatePage.tsx:253]: Yeah, Rob was quite insistent that we should have something there. We populate it initially with the code host URN anyway so the user doesn't have to change it if they don't want to, but it has to have *something*.

Comment (sashaostrikov) [reply client/web/src/site-admin/WebhookCreateUpdatePage.tsx:253]: gotcha, thanks! :)

File: client/web/src/site-admin/SiteAdminWebhookUpdatePage.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                            createdBy={webhook.createdBy}
+                            updatedAt={webhook.updatedAt}
+                            updatedBy={webhook.updatedBy}
+                        />
+                    }
+                    className="mb-3"
+                    headingElement="h2"
+                />
+            )}
+            {webhook && <WebhookCreateUpdatePage existingWebhook={webhook} history={history} />}
```
Comment (mrnugget) [inline client/web/src/site-admin/SiteAdminWebhookUpdatePage.tsx:55]: Nit: you can have one condition and group `PageHeader` and `WebhookCreateUpdatePage` with `<><PageHeader /><WebhookCreateUpdatePage /></>`

File: client/web/src/site-admin/WebhookCreateUpdatePage.tsx (hunk 4 above)
Comment (mrnugget) [reply client/web/src/site-admin/WebhookCreateUpdatePage.tsx:253]: In the schema `name!` is also not optional.

**Review by mrnugget — APPROVED** [review]

**Pull request merged**